renderdoc/driver/vulkan/vk_shaderdebug.cpp

```diff
@@ -5080,6 +5210,87 @@ ShaderDebugTrace *VulkanReplay::DebugThread(uint32_t eventId,
   return ret;
 }
 
+ShaderDebugTrace *VulkanReplay::DebugMeshletThread(uint32_t eventId,
+                                                   const rdcfixedarray<uint32_t, 3> &groupid,
+                                                   const rdcfixedarray<uint32_t, 3> &threadid)
+{
+  const VulkanRenderState &state = m_pDriver->GetRenderState();
+  VulkanCreationInfo &c = m_pDriver->m_CreationInfo;
+
+  rdcstr regionName =
+      StringFormat::Fmt("DebugMeshletThread @ %u of (%u,%u,%u) (%u,%u,%u)", eventId, groupid[0],
+                        groupid[1], groupid[2], threadid[0], threadid[1], threadid[2]);
+
+  VkMarkerRegion region(regionName);
+
+  if(Vulkan_Debug_ShaderDebugLogging())
+    RDCLOG("%s", regionName.c_str());
+
+  const ActionDescription *action = m_pDriver->GetAction(eventId);
+
+  if(!(action->flags & ActionFlags::MeshDispatch))
+  {
+    RDCLOG("No mesh dispatch selected");
+    return new ShaderDebugTrace();
+  }
+
+  // get ourselves in pristine state before this dispatch (without any side effects it may have had)
+  m_pDriver->ReplayLog(0, eventId, eReplay_WithoutDraw);
+
+  const VulkanCreationInfo::Pipeline &pipe = c.m_Pipeline[state.graphics.pipeline];
+  const VulkanCreationInfo::ShaderEntry &shaderEntry =
+      state.graphics.shaderObject ? c.m_ShaderObject[state.shaderObjects[7]].shad : pipe.shaders[7];
```

Comment:
I should fix these plain indices in the other cases, but can you use `[(size_t)ShaderStage::Mesh]` rather than `[7]` here? Since at least I definitely had to double check it was actually the right index.

Reply:
Sure thing. I was just following the existing pattern but using the enum is way more clear I agree